contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,260 @@
+
+    string private constant ERROR_NO_SCRIPT = "DELAY_NO_SCRIPT";
+    string private constant ERROR_CANNOT_FORWARD = "DELAY_CANNOT_FORWARD";
+    string private constant ERROR_CANNOT_EXECUTE = "DELAY_CANNOT_EXECUTE";
+    string private constant ERROR_NOT_ACTIVE = "DELAY_NOT_ACTIVE";
+    string private constant ERROR_SENDER_CANNOT_CANCEL = "DELAY_SENDER_CANNOT_CANCEL";
+
+    enum DelayedScriptStatus {
+        Active,              // A delayed script that has been reported to Agreements
+        Paused,              // A delayed script that is being challenged by Agreements
+        Cancelled,           // A delayed script that has been cancelled
+        Executed             // A delayed script that has been executed
+    }
+
+    struct DelayedScript {
+        uint64 executionFromTime;
+        uint64 pausedAt;
+        DelayedScriptStatus delayedScriptStatus;
+        bytes evmCallScript;
+        uint256 actionId;
+        address submitter;
+    }
+
+    uint64 public executionDelay;
+    uint256 public delayedScriptsNewIndex = 0;
+    mapping (uint256 => DelayedScript) public delayedScripts;
+
+    event ExecutionDelaySet(uint64 indexed executionDelay);
+    event DelayedScriptStored(uint256 indexed delayedScriptId, uint256 indexed actionId, bytes evmScript);
+    event ExecutionPaused(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutionResumed(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutionCancelled(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event AgreementActionClosed(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutedScript(uint256 indexed delayedScriptId, uint256 indexed actionId);
+
+    /**
+    * @notice Initialize the Delay app
+    * @param _executionDelay The delay in seconds a user will have to wait before executing a script
+    */
+    function initialize(uint64 _executionDelay) external onlyInit {
+        initialized();
+        _setExecutionDelay(_executionDelay);
+    }
+
+    /**
+    * @notice Set the execution delay to `_executionDelay`
+    * @param _executionDelay The new execution delay
+    */
+    function setExecutionDelay(uint64 _executionDelay) external auth(SET_DELAY_ROLE) {
+        _setExecutionDelay(_executionDelay);
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function canChallenge(uint256 _delayedScriptId) external view returns (bool) {
+        return _canPause(delayedScripts[_delayedScriptId]);
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function canClose(uint256 _delayedScriptId) external view returns (bool) {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        return _canExecute(delayedScript)
+            || delayedScript.delayedScriptStatus == DelayedScriptStatus.Cancelled;
```

Comment:
If the delayable was canceled the action is automatically closed, so no need to cover that. However, we should consider the case when the delay is already executed. Although this last one is also covered in the `execute` function, but it feels clearer

```suggestion
            || delayedScript.delayedScriptStatus == DelayedScriptStatus.Executed;
```

Reply:
This is necessary for the cancel function to work as seen here: https://github.com/1Hive/disputable-delay-app/pull/1/files#diff-7924fca50b4192b38ad10d5f18cf6a27R214-R216

However, since you've suggested the functionality here: https://github.com/1Hive/disputable-delay-app/pull/1/files#diff-7924fca50b4192b38ad10d5f18cf6a27R189-R192 will be moved to the Agreements app, and there's a chance the action will not close when `_closeExecutionAction()` is called, I will also add `|| delayedScript.delayedScriptStatus == DelayedScriptStatus.Executed;` to `_canExecute()`

Reply:
Looks good! I'll introduce a change in the agreement app to avoid the `canClose` check if the caller is the disputable app itself. I'll let you know!